collector/perf_linux.go

```diff
@@ -0,0 +1,604 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"fmt"
+	"runtime"
+	"sync"
+
+	perf "github.com/hodgesds/perf-utils"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+const (
+	perfSubsystem = "perf"
+
+	// perfCheckMsg is a helper message to return when profilers can't be
+	// started.
+	perfCheckMsg = "ensure that /proc/sys/kernel/perf_event_paranoid is set to 0"
+)
+
+var (
```

Comment:
All these shouldn't be global variables but members of the perfCollector struct.

Reply:
:+1: